hosting/libs/firebase/auth.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  type User,
+  GoogleAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged as _onAuthStateChanged,
+} from "firebase/auth";
+
+import { firebaseAuth } from "./config";
+
+export function onAuthStateChanged(callback: (authUser: User | null) => void) {
+  return _onAuthStateChanged(firebaseAuth, callback);
+}
+
+export async function signInWithGoogle() {
+  const provider = new GoogleAuthProvider();
+
+  try {
+    const result = await signInWithPopup(firebaseAuth, provider);
+
+    if (!result || !result.user) {
+      throw new Error("Google sign in failed");
+    }
+    return result.user.uid;
+  } catch (error) {
+    console.error("Error signing in with Google", error);
+  }
+}
+
+export async function signOutWithGoogle() {
+  try {
+    await firebaseAuth.signOut();
```

Comment:
It would be better if we could handle this success condition and show some message to user with snackbar

Reply:
No need to notify that logout succeeded in a text message. It's enough to redirect out of the platform and back to the sign in page. The message that you're outside is clear.